### PR TITLE
Mcs 171: calc_obj_pos in goals.py is broken

### PR DESCRIPTION
**Pull request by chrislong**
fixed multiple bugs

* missing arguments for calc_obj_coords
* missing arg for point_within_room
* broken test for pickupable-ness and stackTarget-ness
* missing arguments to superclass method call for update_body
* JSON-ification bug

### CODE REVIEW

File: scene_generator/goals.py [hunk 1]
```diff
@@ -496,16 +496,16 @@ def _set_goal_objects(self):
...
         if len(objects) < 2:
             raise ValueError(f'need at least 2 objects for this goal, was given {len(objects)}')
-        target1, target2 = objects
-        print(target2)
-        if 'pickupable' not in target1.get('attributes', []):
+        target1, target2 = objects[0:2]
+        if not target1.get('pickupable', False):
             raise ValueError(f'first object must be "pickupable": {target1}')
-        if 'stackTarget' not in target2.get('attributes', []):
+        if not target2.get('stackTarget', False):
```
Comment (deanwetherby) [inline scene_generator/goals.py:508]: stackTarget and pickupable are part of the attributes field and they're not booleans. This won't work.
objects.py
143    "attributes": ["moveable", "pickupable", "stackTarget"],

Comment (chrislong) [reply scene_generator/goals.py:508]: They're attributes in the object definitions, but they turn into fields when instantiated. See `instantiate_object`. I think it's a bit confusing, I'm not sure why it was decided to be that way.

Comment (deanwetherby) [reply scene_generator/goals.py:508]: Ok. I see it now.
    for attribute in object_def['attributes']:
        new_object[attribute] = True
It is confusing.

File: scene_generator/geometry.py [hunk 2]
```diff
@@ -101,14 +100,20 @@ def calc_obj_pos(performer_position, other_rects, old_object):
 
     dx = old_object['dimensions']['x']/2.0
     dz = old_object['dimensions']['z']/2.0
+    if 'offset' in old_object:
+        offset_x = old_object['offset']['x']
+        offset_z = old_object['offset']['z']
+    else:
+        offset_x = 0.0
+        offset_z = 0.0
```
Comment (ThomasSchellenbergNextCentury) [inline scene_generator/geometry.py:108]: This isn't super important, but can't this be

```
offset_x = old_object['offset']['x'] if 'offset' in old_object else 0.0
offset_z = old_object['offset']['z'] if 'offset' in old_object else 0.0
```

File: scene_generator/goals.py (hunk 1 above)
Comment (ThomasSchellenbergNextCentury) [reply scene_generator/goals.py:508]: If you all think it's less confusing to replace the attributes in the object definitions with separate properties, and just copy each property over into the new_object, that's fine.  Just make a ticket for it.

File: scene_generator/geometry.py (hunk 2 above)
Comment (chrislong) [reply scene_generator/geometry.py:108]: Yeah, that would work, too. Strangely, I usually prefer concision, but I like it the more verbose way this time. But I don't care that much either way.

File: scene_generator/goals.py [hunk 3]
```diff
@@ -496,16 +496,16 @@ def _set_goal_objects(self):
             raise ValueError(f'No stack targets found for transferral goal')
         target2_location = calc_obj_pos(self._performer_start['position'], self._bounding_rects, target2_def)
         target2 = instantiate_object(target2_def, target2_location)
+        logging.debug(f'target2 = {target2}')
```
Comment (brianpippin) [inline scene_generator/goals.py:499]: Do we need this log checked in?

Comment (chrislong) [reply scene_generator/goals.py:499]: I guess not, but the advantage of using logging is that it won't show up unless the user enables log level at least debug.

Comment (brianpippin) [reply scene_generator/goals.py:499]: I guess what I should have asked was is this logging important for logging in the program or was it more of a development debug logging that we didn't need.  I should have been more specific.

Comment (chrislong) [reply scene_generator/goals.py:499]: I'm not sure. This PR has been merged, so I'll take it out in the branch I'm working on now (MCS-118).

**Pull request merged**